pkg/grafana/client_test.go

```diff
@@ -82,39 +83,23 @@ func TestUpsertOrgUser(t *testing.T) {
 
 // This is a silly test as the mock always returns nil but it's here for completeness
 func TestUpdateUserPermissions(t *testing.T) {
-	user := setupUser()
+	user := newUser("foo", 1)
 
 	client := NewMockClient(user, userOrgsRoleMap{})
 
 	err := client.UpdateUserPermissions(user.ID, true)
 	assert.NoError(t, err)
 }
 
-func TestUpdateUserGrafanaAdmin(t *testing.T) {
-	user := setupUser()
-
-	client := NewMockClient(user, userOrgsRoleMap{})
-
-	err := client.updateUserGrafanaAdmin(user, true)
-	assert.NoError(t, err)
-
-	err = client.updateUserGrafanaAdmin(user, false)
-	assert.NoError(t, err)
-}
-
 func TestUpdateOrgUserAuthz(t *testing.T) {
```

Comment:
I think this needs test cases for:

* user is Admin and 1 group in N groups is a grafana admin
** c.UpdateUserPermissions(user.ID, isGlobalAdmin) is not called
* user is not Admin and 1 group in N groups is a grafana admin
** c.UpdateUserPermissions(user.ID, isGlobalAdmin) is called with isGlobalAdmin true
* user is Admin and no group is a grafana admin
** c.UpdateUserPermissions(user.ID, isGlobalAdmin) is called with isGlobalAdmin false

Reply:
See my reply on the previous comment. Will try to find a way to retrieve the updated user and compare

Reply:
done that, now the intention is more clear as the mock results are shown directly on the tests